Neo4j.Driver/Neo4j.Driver.Tests/Result/PeekingEnumeratorTests.cs

```diff
@@ -115,15 +115,32 @@ public void ShouldConsumeAllItemsAfterDiscard()
         public class DisposeMethod
         {
             [Fact]
-            public void ShouldNotDiscardAfterDispose()
+            public void ShouldDiscardAfterDispose()
             {
                 var list = new List<string> { "1", "2", "3" }.GetEnumerator();
                 var enumerator = new PeekingEnumerator<string>(list);
-                enumerator.MoveNext().Should().BeTrue();
-                enumerator.Consume();
+                enumerator.Dispose();
                 enumerator.MoveNext().Should().BeFalse();
                 enumerator.Peek().Should().BeNull();
-                enumerator.Position.Should().Be(3);
+                enumerator.Position.Should().Be(-1);
+            }
+
+            [Fact]
+            public void PeakShouldReturnDefaultAfterDispose()

```

Comment:
Sorry, I think my changeset is ok. I have made the code act the same way as other enumerators in the standard .NET framework. They either throws or return frfault values. 

I think that divegating from standards can lead to bugs. 

We can leave out this commit and fix thr warning another way?